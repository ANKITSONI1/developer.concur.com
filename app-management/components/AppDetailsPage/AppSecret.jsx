import React, { PropTypes } from 'react';
import LoadingSpinner from '../LoadingSpinner';
import ErrorAlert from '../ErrorAlert';

const AppSecret = ({ clickHandler, appSecret: { clientSecret, error, isFetching } }) => (
  <div>
    <div className="col-md-8">
      <a href="#client-secret-generation" data-toggle="collapse">Regenerate clientSecret</a>
      <div
        className="collapse panel panel-danger"
        data-toggle="collapse"
        id="client-secret-generation"
      >
        <div className="panel-body">
          <ErrorAlert error={error} />
          <LoadingSpinner loading={isFetching} />
          {clientSecret ? (
            <p>
              <strong>
                Here is your new clientSecret. Make sure to copy it now. You will not be able
                to see it again.
              </strong>
              <br />
              <code>{clientSecret}</code>
            </p>
          ) : (
            <div>
              <p>
                You can regenerate your clientSecret below.
              </p>
              <p>
                <strong>WARNING: </strong>
                This process will clear your old clientSecret and require updates to your
                apps or scripts.
              </p>
              <button
                type="button"
                className="btn orange small"
                onClick={clickHandler}
              >
                Regenerate clientSecret
              </button>
            </div>
          )}
        </div>
      </div>
    </div>
  </div>
);

AppSecret.propTypes = {
  clickHandler: PropTypes.func.isRequired,
  appSecret: PropTypes.shape({
    clientSecret: PropTypes.string.isRequired,
    error: PropTypes.string.isRequired,
    isFetching: PropTypes.bool.isRequired,
  }).isRequired,
};

export default AppSecret;
