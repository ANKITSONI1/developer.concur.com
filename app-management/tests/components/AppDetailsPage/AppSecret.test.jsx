import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';

import AppSecret from '../../../components/AppDetailsPage/AppSecret';

describe('<AppSecret />', () => {
  const defaultProps = {
    clickHandler: jest.fn(),
    appSecret: {
      clientSecret: '',
      error: '',
      isFetching: false,
    }
  };

  it('should call the click handler when "Get New Secret" is pressed', () => {
    const secret = shallow(<AppSecret {...defaultProps} />);
    secret.find('button').simulate('click');

    expect(defaultProps.clickHandler).toHaveBeenCalled();
  });

  it('should not display the secret if not provided', () => {
    const secret = shallow(<AppSecret {...defaultProps} />);

    expect(shallowToJson(secret)).toMatchSnapshot();
  });

  it('should display the secret if provided', () => {
    const props = {
      ...defaultProps,
      appSecret: {
        ...defaultProps.appSecret,
        clientSecret: 'a-test-secret',
      },
    };
    const secret = shallow(<AppSecret {...props} />);

    expect(shallowToJson(secret)).toMatchSnapshot();
  });
});
