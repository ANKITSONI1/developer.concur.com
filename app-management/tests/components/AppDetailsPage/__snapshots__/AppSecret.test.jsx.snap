// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AppSecret /> should display the secret if provided 1`] = `
<div>
  <div
    className="col-md-8"
  >
    <a
      data-toggle="collapse"
      href="#client-secret-generation"
    >
      Regenerate clientSecret
    </a>
    <div
      className="collapse panel panel-danger"
      data-toggle="collapse"
      id="client-secret-generation"
    >
      <div
        className="panel-body"
      >
        <ErrorAlert
          error=""
        />
        <LoadingSpinner
          loading={false}
          message="Loading..."
        />
        <p>
          <strong>
            Here is your new clientSecret. Make sure to copy it now. You will not be able to see it again.
          </strong>
          <br />
          <code>
            a-test-secret
          </code>
        </p>
      </div>
    </div>
  </div>
</div>
`;

exports[`<AppSecret /> should not display the secret if not provided 1`] = `
<div>
  <div
    className="col-md-8"
  >
    <a
      data-toggle="collapse"
      href="#client-secret-generation"
    >
      Regenerate clientSecret
    </a>
    <div
      className="collapse panel panel-danger"
      data-toggle="collapse"
      id="client-secret-generation"
    >
      <div
        className="panel-body"
      >
        <ErrorAlert
          error=""
        />
        <LoadingSpinner
          loading={false}
          message="Loading..."
        />
        <div>
          <p>
            You can regenerate your clientSecret below.
          </p>
          <p>
            <strong>
              WARNING: 
            </strong>
            This process will clear your old clientSecret and require updates to your apps or scripts.
          </p>
          <button
            className="btn orange small"
            onClick={[Function]}
            type="button"
          >
            Regenerate clientSecret
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;
